### ISSUE TITLE
Add an "Extending Flux" page to the docs

### ISSUE THREAD

**Issue by logankilpatrick** [FluxML/Flux.jl#1811]
At the beginning of the docs, we claim that Flux is extendable and Flexible yet we don't really go into why that is the case. We should have this in the docs. PyTorch for example has: https://pytorch.org/docs/stable/notes/extending.html

**Comment by ToucheSir** [comment]
I agree, currently we have scattered bits and pieces like the functors-related docs, but nothing coherent. One challenge is finding the right balance between what is "extending Flux" and what is "normal Julia programming". One of our selling points is that one doesn't have to learn a whole new set of concepts to use the library, after all. Perhaps a start would be just focusing on what Flux itself offers in-repo (layers, maybe losses) and delegating to other documentation such as the excellent https://juliadiff.org/ChainRulesCore.jl/stable/.

**Comment by CarloLucibello** [comment]
Closing — this is now covered by the custom layers tutorial (`docs/src/tutorials/custom_layers.md`) plus the `@layer`/functors reference page, which together document what is needed to extend Flux with your own layers.
